picard/ui/options/metadata.py

```diff
@@ -92,6 +92,7 @@ class MetadataOptionsPage(OptionsPage):
         ListOption('setting', 'script_exceptions', [], title=N_("Translation script exceptions")),
         BoolOption('setting', 'release_ars', True, title=N_("Use release relationships")),
         BoolOption('setting', 'track_ars', False, title=N_("Use track and release relationships")),
+        BoolOption('setting', 'disable_date_sanitize', False, title=N_("Disable date sanitization for APE and Vorbis tags")),
```

Comment:
As discussed, it would be better to have a list of formats for which we want to disable date sanitization.
It can be any format NAME has reported by https://github.com/metabrainz/picard/blob/16b2ddc9472a294ad636fd7858f731d724eb2af7/picard/formats/util.py#L43

And this `BoolOption` can be converted to a set:

```python
Option('setting', 'disable_date_sanitize', set(), title=N_("Disable date sanitization for specified formats")),
```

Of course a selector should let the user select multiple formats.

We could use a multiselect combobox (see, for example, https://stackoverflow.com/a/77755095), and add the widget to https://github.com/metabrainz/picard/tree/master/picard/ui/widgets for reuse.